### PR TITLE
chapter11_account_activation

### PR DESCRIPTION
**Pull request by tenyiting**
chapter11_account_activation

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

**Comment by tenyiting** [comment]
ready

File: app/assets/javascripts/account_activations.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (ghost) [inline app/assets/javascripts/account_activations.coffee]: please delete this file

File: app/assets/stylesheets/account_activations.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
```
Comment (ghost) [inline app/assets/stylesheets/account_activations.scss]: please delete this file

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
 class UsersController < ApplicationController
-  before_action :logged_in_user, except: %i(new show create)
-  before_action :correct_user, only: %i(edit update)
   before_action :load_user, only: :show
+  before_action :logged_in_user, only: %i(index edit update destroy)
+  before_action :correct_user, only: %i(edit update)
   before_action :admin_user, only: %i(destroy)
+  skip_before_action :verify_authenticity_token
```
Comment (ghost) [inline app/controllers/users_controller.rb]: why you have to add this line?

File: app/helpers/account_activations_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (ghost) [inline app/helpers/account_activations_helper.rb]: please delete this file

File: app/mailers/user_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  # Subject can be set in your I18n file at config/locales/en.yml
+  # with the following lookup:
+  #
+  #   en.user_mailer.password_reset.subject
+  #
+
+  def account_activation(user)
+    @user = user
+    mail to: user.email, subject: "Account activation"
```
Comment (ghost) [inline app/mailers/user_mailer.rb]: i18n please

File: app/views/user_mailer/account_activation.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<h1><%= t(:h_title) %></h1>
+
+<p><%= t(:hi) %> <%= @user.name %> <%= t(:comma) %></p>
+<p> <%= t(:w_activate_acc) %> </p>
+
+<%= link_to "Activate", edit_account_activation_url(id:
```
Comment (ghost) [inline app/views/user_mailer/account_activation.html.erb]: i18n please

File: test/mailers/user_mailer_test.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+require 'test_helper'
```
Comment (ghost) [inline test/mailers/user_mailer_test.rb]: You don't need to do test
please delete this file

File: test/mailers/previews/user_mailer_preview.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
```
Comment (ghost) [inline test/mailers/previews/user_mailer_preview.rb]: please delete this file

File: test/controllers/account_activations_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
```
Comment (ghost) [inline test/controllers/account_activations_controller_test.rb]: please delete this file

**Review by ghost — COMMENTED** [review]
cmt

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**